homeassistant/components/intent_script/__init__.py

```diff
@@ -144,7 +144,11 @@ async def async_handle(self, intent_obj: intent.Intent):
                     action.async_run(slots, intent_obj.context)
                 )
             else:
-                await action.async_run(slots, intent_obj.context)
+                action_res = await action.async_run(slots, intent_obj.context)
+
+                # if the action returns a response, make it available to the speech/reprompt templates below
+                if action_res.service_response is not None:
+                    slots.update(action_res.service_response)
```

Comment:
How do we know that there is no overlap between the response value keys and the slot keys?

Reply:
I just thought that overlaps are rare, and there is always the easy workaround of "restructuring" the response from within the action. Eg if there is a slot named `events` you can easily write.
```yaml
intent_script:
  EventCountToday:
    action:
      - service: calendar.list_events
        target:
          entity_id: calendar.my_calendar
        data_template:
          start_date_time: "{{ today_at('00:00') }}"
          duration: { "hours": 24 }
        response_variable: result         # get a response from the service
      - variables:
          result2:                        # restructure
            events2: "{{ result.events }}"
      - stop: ""
        response_variable: result2        # and then return
    speech:
      text: "{{ events2 | length }}"      # no overlap with the slot
```

So the common case is simple, and the rare case is still doable with a small workaround.

Another approach would be to introduce another `response_variable` at the level of `intent_script`:
```yaml
intent_script:
  EventCountToday:
    action:
      - service: calendar.list_events
        target:
          entity_id: calendar.my_calendar
        data_template:
          start_date_time: "{{ today_at('00:00') }}"
          duration: { "hours": 24 }
        response_variable: result         # get a response from the service
      - stop: ""
        response_variable: result         # and return
    response_variable: my_var
    speech:
      text: "{{ mt_var.events | length }}"      # no overlap with the slot
```

Maybe this is more consistent, but the double `response_variable` is cumbersome.

Reply:
We've discussed this PR in the core team. We suggest we hardcode a response variable. I suggest naming it `response_result`.

Reply:
Ok. Maybe `action_response` is better to make it clear that it comes from the `action`?

Reply:
Done, let me know if you need anything else.